tests/test_packageurl.py

```diff
@@ -330,3 +330,47 @@ def test_to_dict_custom_empty_value(self):
 def test_purl_is_hashable():
     s = {PackageURL(name="hashable", type="pypi")}
     assert len(s) == 1
+
+
+def test_colons_in_name_are_handled_correctly() -> None:
+    p = PackageURL.from_string(
+        "pkg:nuget/libiconv:%20character%20set%20conversion%20library@1.9?package-id=e11a609df352e292"
+    )
+
+    assert p.type == "nuget"
+    assert p.namespace is None
+    assert p.name == "libiconv: character set conversion library"
+    assert p.version == "1.9"
+    assert p.qualifiers == {"package-id": "e11a609df352e292"}
+    assert p.subpath == None
+
+    assert PackageURL.from_string(p.to_string()).to_string() == p.to_string()
+
+
+def test_colons_in_namespace_are_handled_correctly() -> None:
+    p = PackageURL.from_string(
+        "pkg:nuget/an:odd:space/libiconv:%20character%20set%20conversion%20library@1.9?package-id=e11a609df352e292"
```

Comment:
Could you use actual real life data rather than made up ones? In all cases the colon should be encoded there IMHO.

Reply:
I haven't seen a `:` in a namespace, but just in a name. I used a real-life name in this test that I had found in an SBOM that was given to us.

Reply:
Ok, so you saw a colon in a namespace in a PURL? I would really like to see that exact PURL and which tool did create that.

Reply:
I did not have a namespace with a colon, but I *did* see a name with a colon.

Reply:
@sjn you have :: colons in CPAN, do you?